### ISSUE TITLE
[date-picker] focus state on calendar dropdown clashes with selected date

### ISSUE THREAD

**Issue by dakahn** [carbon-design-system/carbon#5308]
Give the date picker component keyboard focus and hit the down arrow to open the calendar dropdown. Focus has moved to the dropdown component, but if there is a previously selected date visually the focus state is undetectable
![The selected date and the currently focused date is 2/5](https://user-images.githubusercontent.com/40970507/74171279-f05d3d80-4bf3-11ea-849c-1bd34b39efa9.png)
_The selected date and the currently focused date is 2/5_

@snidersd tagging you here since you found this one

**Comment by snidersd** [comment]
Testing this with VO, macOS and Chrome with the following issues:
1. After a date is selected from the calendar VO announces that it is invalid data. This happens in all date picker calendars.
<img width="1440" alt="range-calendar" src="https://user-images.githubusercontent.com/21676914/74190706-67a4c880-4c18-11ea-98f5-d97875f689a9.png">
2. Date in the calendar should be selectable with Space and Enter key and it should close the dialog after the date is selected. Space does not work and the user must hit the Enter key twice to close the dialog. In addition, the second time the Enter key is hit it also populates the second date in the date range.

3. Range calendar with min/max - displays the January calendar when it opens and does not display the min/max range until the second date picker is selected. 
<img width="1047" alt="Screen Shot 2020-02-10 at 3 07 36 PM" src="https://user-images.githubusercontent.com/21676914/74191562-763faf80-4c19-11ea-81fc-96d29561692c.png">
4. Fully controlled - 
Arrow down to dates and VO starts announcing every date in the calendar.
<img width="1172" alt="Screen Shot 2020-02-10 at 3 10 22 PM" src="https://user-images.githubusercontent.com/21676914/74191637-996a5f00-4c19-11ea-8b30-8d9b0c3af488.png">

Here is a link to the WAI-ARIA date picker pattern for keyboard accessible keyboard controls: https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/datepicker-dialog.html

**Comment by aagonzales** [comment]
Good catch, for selected+focus lets treat it the same we do with our primary button, so it should get that 1px white inset border.

![image](https://user-images.githubusercontent.com/11670886/82072981-28b26300-969e-11ea-87d4-4753afab867f.png)
